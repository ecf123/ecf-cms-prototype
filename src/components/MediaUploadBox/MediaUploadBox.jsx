import "./MediaUploadBox.scss";
import bin from "../../assets/bin-icon.svg";
import Button from "../Button/Button";

const MediaUploadBox = ({
  file,
  handleFileChange,
  uploadLabelName = "Pathway Thumbnail",
  uploadButtonText = "Upload Image",
  fileName,
}) => {
  return (
    <div className="thumbnail">
      <label htmlFor="file-input" className="thumbnail__label">
        {uploadLabelName}
      </label>
      <div className="thumbnail__container">
        {!file && (
          <div className="thumbnail__btn--wrapper">
            < Button style={"button light-grey round-border small"}  textStyle={"text small-text light-grey"}  buttonText={uploadButtonText}/>
            <input
              className="thumbnail__btn--input"
              id="file-input"
              type="file"
              onChange={handleFileChange}
              accept="image/*,video/*,audio/*"
            />
          </div>
        )}
        {file && (
          <img src={file} alt={fileName} className="thumbnail__picture" />
        )}
      </div>
      {file && (
        <div className="thumbnail__remove" onClick={handleFileChange}>
          <img className="thumbnail__remove--bin" src={bin} alt="bin icon" />
          <p className="thumbnail__remove--text">Remove</p>
        </div>
      )}
    </div>
  );
};

export default MediaUploadBox;
