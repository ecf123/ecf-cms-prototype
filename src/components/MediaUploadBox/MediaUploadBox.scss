@use "../../styles/variables" as *;
@use "../../styles/main.scss" as *;

.thumbnail {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  gap: 0.375rem;

  width: 19.5rem;
  height: 16.25rem;

  font-family: $font-secondary;

  &__container {
    box-sizing: border-box;

    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    gap: 0.5rem;

    width: 19.5rem;
    height: 12.5rem;

    background: $color-white;
    border: 1px dashed #d0d5dd;
    box-shadow: 0px 1px 2px rgba(16, 24, 40, 0.05);
    border-radius: 8px;
  }

  &__picture {
    max-width: 90%;
    max-height: 90%;
  }

  &__label {
    font-size: 0.875rem;
    line-height: 1.25rem;
    color: $color-blue-navy;
  }

  &__remove {
    display: flex;
    align-items: center;
    justify-content: space-between;
    cursor: pointer;
    background-color: transparent;
    border: none;

    &--bin {
      margin: 0 0.5rem;
      filter: invert(71%) sepia(88%) saturate(6543%) hue-rotate(336deg)
        brightness(98%) contrast(86%);
    }

    &--text {
      font-weight: 300;
      font-size: $font-size-smallest;
      line-height: 0.75rem;
      color: $color-red-light;
    }
  }

  &__btn {
    &--wrapper {
      position: relative;
      overflow: hidden;
      display: inline-block;
    }

    &--input {
      font-size: $font-size-extra-large;
      position: absolute;
      left: 0;
      top: 0;
      opacity: 0;
      cursor: pointer;
    }
  }
}
