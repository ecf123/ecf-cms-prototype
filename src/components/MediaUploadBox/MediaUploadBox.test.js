import MediaUploadBox from "./MediaUploadBox";
import {render, screen} from "@testing-library/react";
import rollercoasterimg from "../../assets/rollercoaster-image.svg";
import userEvent from '@testing-library/user-event';

const fileImage = rollercoasterimg;
describe(MediaUploadBox, () => {
    // it("should display the picture chosen by the user on the screen", () => {
    //     render(<MediaUploadBox file={fileImage}/>) 
    //     // const inputImg = getByRole("input", {type: "file"});
    //     // userEvent.click(inputImg);
    //     const paragraph = screen.getByRole("paragraph");

    //     expect(paragraph).toBeInTheDocument();
    // });

    it("should display thumbnail on render", () => {
        render(<MediaUploadBox/>);
        const label = screen.getByLabelText("Pathway Thumbnail");

        expect(label).toBeTruthy();
    })
    it("should display the upload button on render", () => {
        render(<MediaUploadBox/>);
        const button = screen.getByRole("button");

        expect(button).toBeInTheDocument()
    })
    // it("should contain input in the document", () => {
    //     render(<MediaUploadBox/>);
    //     const input = screen.getByText("Remove");

    //     expect(input).not.toBeVisible();
    // })

})

// test(MediaUploadBox, () => {
//     it("should remove the selected image if remove button is clicked", () => {
//         const {} = render(<MediaUploadBox/>)

//         expect().toBe();
//     });

// })