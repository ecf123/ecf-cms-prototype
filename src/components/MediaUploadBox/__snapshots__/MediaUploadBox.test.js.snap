// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaUploadBox component matches the snapshot taken with the file 1`] = `
<div>
  <div
    class="thumbnail"
  >
    <label
      class="thumbnail__label"
      for="file-input"
    >
      Pathway Thumbnail
    </label>
    <div
      class="thumbnail__container"
    >
      <img
        class="thumbnail__picture"
        src="rollercoaster-image.svg"
      />
    </div>
    <div
      class="thumbnail__remove"
    >
      <img
        alt="bin icon"
        class="thumbnail__remove--bin"
        src="bin-icon.svg"
      />
      <p
        class="thumbnail__remove--text"
      >
        Remove
      </p>
    </div>
  </div>
</div>
`;

exports[`MediaUploadBox component matches the snapshot taken without the file 1`] = `
<div>
  <div
    class="thumbnail"
  >
    <label
      class="thumbnail__label"
      for="file-input"
    >
      Pathway Thumbnail
    </label>
    <div
      class="thumbnail__container"
    >
      <div
        class="thumbnail__btn--wrapper"
      >
        <button
          class="button light-grey round-border small"
        >
          <div />
          <p
            class="text small-text light-grey"
          >
             
            Upload Image
          </p>
        </button>
        <input
          accept="image/*,video/*,audio/*"
          class="thumbnail__btn--input"
          id="file-input"
          type="file"
        />
      </div>
    </div>
  </div>
</div>
`;
